vignettes/decorate-module-output.Rmd

```diff
@@ -0,0 +1,589 @@
+---
+title: "Customizing Module Output"
+author: "NEST CoreDev"
+output:
+  rmarkdown::html_vignette:
+    toc: true
+vignette: >
+  %\VignetteIndexEntry{Customizing Module Output}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+## Introduction
+
+`teal` is a powerful `shiny`-based framework with built-in modules for interactive data analysis. 
+This document outlines the customization options available for modifying the output of `teal` modules. 
+You will learn how to use `teal_transform_module` to modify and enhance the objects created by `teal` modules, 
+enabling you to tailor the outputs to your specific requirements without rewriting the original module code.
+
+Adjusting input data or customizing module outputs in `teal` is accomplished using `transformators` created through 
+`teal_transform_module`. This document will focus on modifying outputs, and for clarity, we will refer to these 
+output-modifying objects as decoration objects or decorators.
+
+
+## Decorators
+
+### Server
+
+The simplest way to create a decorator is to use `teal_transform_module` without specifying a `ui`.
+This approach adds functionality solely to the server code of the `teal` app. 
+In the following example, we assume that the module contains an object (of class `ggplot2`) named `plot`. 
+We modify the title and x-axis label of plot:
+
+```{r}
+library(teal)
```

Comment:
you can move library to a separate hidden code chunk upfront

Reply:
I'll keep it as we also display `library(teal)` in different vignettes.